File: src/main/java/org/example/kalkulationsprogramm/dto/Mitarbeiter/MitarbeiterDto.java
package org.example.kalkulationsprogramm.dto.Mitarbeiter;

import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Data
public class MitarbeiterDto {
    private Long id;
    private String vorname;
    private String nachname;
    private String strasse;
    private String plz;
    private String ort;
    private String email;
    private String telefon;
    private String festnetz;
    private String qualifikation;
    private BigDecimal stundenlohn;
    private Integer jahresUrlaub;
    private LocalDate geburtstag;
    private LocalDate eintrittsdatum;
    private Boolean aktiv;
    private List<Long> abteilungIds;
    private String abteilungNames; // Komma-separierte Namen für Anzeige
    private String loginToken;

    // Lohn-/Sozialversicherungs-Felder (siehe V297__mitarbeiter_sv_felder.sql).
    private String beschaeftigungsart;
    private String beschaeftigungsartLabel;
    private Long krankenkasseId;
    private String krankenkasseName;
    private Boolean kinderlos;

    // Geschaeftsfuehrer-Felder (siehe V300__mitarbeiter_geschaeftsfuehrer.sql).
    // Wenn istGeschaeftsfuehrer=true, fliesst kalkulatorischerLohnMonat statt
    // echtes Brutto in die Lohnsumme des Verrechnungslohn-Rechners ein.
    private Boolean istGeschaeftsfuehrer;
    private BigDecimal kalkulatorischerLohnMonat;
    private BigDecimal geldwertVorteilMonat;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/dto/Mitarbeiter/MitarbeiterDto.kt
package org.example.kalkulationsprogramm.dto.Mitarbeiter

import java.math.BigDecimal
import java.time.LocalDate

data class MitarbeiterDto(
    var id: Long? = null,
    var vorname: String? = null,
    var nachname: String? = null,
    var strasse: String? = null,
    var plz: String? = null,
    var ort: String? = null,
    var email: String? = null,
    var telefon: String? = null,
    var festnetz: String? = null,
    var qualifikation: String? = null,
    var stundenlohn: BigDecimal? = null,
    var jahresUrlaub: Int? = null,
    var geburtstag: LocalDate? = null,
    var eintrittsdatum: LocalDate? = null,
    var aktiv: Boolean? = null,
    var abteilungIds: List<Long>? = null,
    var abteilungNames: String? = null,
    var loginToken: String? = null,
    var beschaeftigungsart: String? = null,
    var beschaeftigungsartLabel: String? = null,
    var krankenkasseId: Long? = null,
    var krankenkasseName: String? = null,
    var kinderlos: Boolean? = null,
    var istGeschaeftsfuehrer: Boolean? = null,
    var kalkulatorischerLohnMonat: BigDecimal? = null,
    var geldwertVorteilMonat: BigDecimal? = null,
)
